### PR TITLE
Add template on how to add / modify a project

### LINKED ISSUE #39: Create a markdown contain standardized .yaml file for projects

### PR DESCRIPTION
**Pull request by fiennyangeln**
Fix #39

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
 This project welcomes contributors. If you are interested, please feel free to
 join [the mailing list](https://mail.mozilla.org/listinfo/bugsahoy-devel)
 
+## Add a project
```
Comment (helfi92) [inline README.md]: Put this before the Contributing section. I would also rename to `## Adding a project`

File: README.md [hunk 2]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+
+Each of the projects is a file in the [data](https://github.com/mozilla-frontend-infra/codetribute/tree/master/src/data)
+folder - to add a new one, create a new file named after the project, ending in `.yaml`. Ensure all spaces and special
+characters are replaced with `-`. Make sure too that the project hasn't already been inside the [data](https://github.com/mozilla-frontend-infra/codetribute/tree/master/src/data)
+folder
+
+The contents of the file are just some details about the project:
```
Comment (helfi92) [inline README.md]: How about:

Codetribute can read from both GitHub and Bugzilla. To add a new entry to the site, create a file `<project-name>.yml` in `src/data` using the [template example](#template-example) as the initial setup. For inspiration, check out the [Taskcluster](https://github.com/mozilla-frontend-infra/codetribute/blob/master/src/data/taskcluster.yaml) yaml file.

File: README.md [hunk 3]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+
+```yaml
+name: [project name]
```
Comment (helfi92) [inline README.md]: `name: <projectName>`

File: README.md [hunk 4]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+summary: [a brief description of the project]
```
Comment (helfi92) [inline README.md]: summary: A short summary of the project.

File: README.md [hunk 5]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+[ The introduction contains more information about the projects in markdown format.
```
Comment (helfi92) [inline README.md]: Remove this line.

File: README.md [hunk 6]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+Below are the suggestions on how to write the introduction. ]
```
Comment (helfi92) [inline README.md]: Remove this line.

File: README.md [hunk 7]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+introduction: |
+  ## About [project name]
```
Comment (helfi92) [inline README.md]: `s/[project name]/<Project Name>/`

File: README.md [hunk 8]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+
+  ...
+
+  ## Who Works on [project name]?
```
Comment (helfi92) [inline README.md]: `s/[project name]/<Project Name>/`

File: README.md [hunk 9]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+
+  ...
+
+  ## How Do I Get Help?
+
+  ...
+
+products:
+- [bugzilla product] OR
+- [bugzilla product] : [bugzilla component, specify this if it is not going to be for all component]
```
Comment (helfi92) [inline README.md]:
```
products:
 - <Bugzilla Product>
 - <Bugzilla Product>: [<Bugzilla Componet 1>, <Bugzilla Component 2>]
```

File: README.md [hunk 10]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+repositories:
+- [repository name] : [issues' tag / label to get the bug]
+[
+    if there is more than one tag for a repository, list them in a different line, e.g
+    - taskcluster/taskcluster-queue: good-first-bug
+    - taskcluster/taskcluster-queue: easy-fix
+]
```
Comment (helfi92) [inline README.md]:
```
repositories:
 - <Organization Name>/<Repository Name>: good-first-bug
 - <Organization Name>/<Repository Name>: easy-fix
```

File: README.md [hunk 11]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+```
+
+Check out the [Taskcluster](https://github.com/mozilla-frontend-infra/codetribute/blob/master/src/data/taskcluster.yaml)
+file for an example of this project structure.
```
Comment (helfi92) [inline README.md]: We are going to refer to this file before showing the template file. We can remove this.

File: README.md [hunk 12]
```diff
@@ -62,4 +62,53 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+## Add a project
+
+Each of the projects is a file in the [data](https://github.com/mozilla-frontend-infra/codetribute/tree/master/src/data)
+folder - to add a new one, create a new file named after the project, ending in `.yaml`. Ensure all spaces and special
+characters are replaced with `-`. Make sure too that the project hasn't already been inside the [data](https://github.com/mozilla-frontend-infra/codetribute/tree/master/src/data)
+folder
+
+The contents of the file are just some details about the project:
+
+```yaml
```
Comment (helfi92) [inline README.md]: Add `### Template Example` before the example.

File: README.md [hunk 13]
```diff
@@ -57,31 +57,28 @@ Install npm dependencies and start it up:
...
-This project welcomes contributors. If you are interested, please feel free to
-join [the mailing list](https://mail.mozilla.org/listinfo/bugsahoy-devel)
-
-## Add a project
+## Adding a project
 
 Each of the projects is a file in the [data](https://github.com/mozilla-frontend-infra/codetribute/tree/master/src/data)
 folder - to add a new one, create a new file named after the project, ending in `.yaml`. Ensure all spaces and special
 characters are replaced with `-`. Make sure too that the project hasn't already been inside the [data](https://github.com/mozilla-frontend-infra/codetribute/tree/master/src/data)
 folder
```
Comment (helfi92) [inline README.md]: This still needs to be removed. It's covered by the paragraph right under it.

File: README.md [hunk 14]
```diff
@@ -57,6 +57,57 @@ Install npm dependencies and start it up:
...
+
+  ...
+
+  ## Who Works on <Project Name>?
+
+  ...
+
+  ## How Do I Get Started?
+
+  ...
```
Comment (djmitche) [inline README.md]: (general advice - tutorials to learn about the project, development setup, repo to clone if there is only one)

File: README.md [hunk 15]
```diff
@@ -57,6 +57,57 @@ Install npm dependencies and start it up:
...
+
+  ### How Do I Write the Code?
+
+  ...
```
Comment (djmitche) [inline README.md]: (summary of the development and patch-submission process -- pull requests? patches on bugzilla? tests?)

File: README.md [hunk 16]
```diff
@@ -57,6 +57,57 @@ Install npm dependencies and start it up:
...
+
+  ## How Do I Get Help?
+
+  ...
```
Comment (djmitche) [inline README.md]: (suggestions for how, and when, to ask for help -- mailing lists, irc channels, bug or issue comments, etc.)

File: README.md [hunk 17]
```diff
@@ -59,14 +59,10 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
-Each of the projects is a file in the [data](https://github.com/mozilla-frontend-infra/codetribute/tree/master/src/data)
-folder - to add a new one, create a new file named after the project, ending in `.yaml`. Ensure all spaces and special
-characters are replaced with `-`. Make sure too that the project hasn't already been inside the [data](https://github.com/mozilla-frontend-infra/codetribute/tree/master/src/data)
-folder
-
 Codetribute can read from both GitHub and Bugzilla. To add a new entry to the site, create a file `<project-name>.yml` 
-in `src/data` using the [template example](#template-example) as the initial setup. For inspiration, check out the 
-[Taskcluster](https://github.com/mozilla-frontend-infra/codetribute/blob/master/src/data/taskcluster.yaml) yaml file.
+in `src/data` using the [template example](#template-example) as the initial setup. Ensure all spaces and special
+characters are replaced with `-`. Make sure that the project is not already in the `src/data` folder.
```
Comment (helfi92) [inline README.md]: Remove "Ensure all spaces and special characters are replaced with -. Make sure that the project is not already in the src/data folder."

File: README.md [hunk 18]
```diff
@@ -59,14 +59,10 @@ This will start a local development server on port 5000 (http://localhost:5000).
...
+For inspiration, check out the [Taskcluster](https://github.com/mozilla-frontend-infra/codetribute/blob/master/src/data/taskcluster.yaml) yaml file.
 
 ### Template Example
```
Comment (helfi92) [inline README.md]: ### Template Example

The `introduction` field is to be rendered as markdown, allowing for bullet points, links and other simple formatting.

File: README.md [hunk 19]
```diff
@@ -76,27 +72,29 @@ summary: A short summary of the project
...
 
-  ...
+  <Summary of the development and patch-submission process -- pull requests? patches on bugzilla? tests?>
 
   ## How Do I Get Help?
 
-  ...
+  <Suggestions for how, and when, to ask for help -- mailing lists, irc channels, bug or issue comments, etc.>
+  
+  NB: introduction is written in Markdown.
```
Comment (helfi92) [inline README.md]: Remove this line.

File: README.md [hunk 20]
```diff
@@ -57,6 +57,53 @@ Install npm dependencies and start it up:
...
+
+  ## How Do I Get Help?
+
+  <Suggestions for how, and when, to ask for help -- mailing lists, irc channels, bug or issue comments, etc.>
+
+
+products:
+ - <Bugzilla Product>
+ - <Bugzilla Product>: [<Bugzilla Component 1>, <Bugzilla Component 2>]
+ repositories:
```
Comment (helfi92) [inline README.md]: Fix indentation.

File: README.md [hunk 21]
```diff
@@ -57,6 +57,53 @@ Install npm dependencies and start it up:
...
+- <Organization Name>/<Repository Name> : <Label to get the issue>
+[
```
Comment (helfi92) [inline README.md]: Remove `[`.

File: README.md [hunk 22]
```diff
@@ -57,6 +57,53 @@ Install npm dependencies and start it up:
...
+    if there is more than one tag for a repository, list them in a different line, e.g
```
Comment (helfi92) [inline README.md]: Remove this line.

File: README.md [hunk 23]
```diff
@@ -57,6 +57,53 @@ Install npm dependencies and start it up:
...
+      repositories:
+      - <Organization Name>/<Repository Name>: good-first-bug
+      - <Organization Name>/<Repository Name>: easy-fix
+]
```
Comment (helfi92) [inline README.md]: Remove `]`.

File: README.md [hunk 24]
```diff
@@ -57,6 +57,53 @@ Install npm dependencies and start it up:
...
+products:
+ - <Bugzilla Product>
+ - <Bugzilla Product>: [<Bugzilla Component 1>, <Bugzilla Component 2>]
+ repositories:
+- <Organization Name>/<Repository Name> : <Label to get the issue>
+[
+    if there is more than one tag for a repository, list them in a different line, e.g
+      repositories:
+      - <Organization Name>/<Repository Name>: good-first-bug
+      - <Organization Name>/<Repository Name>: easy-fix
```
Comment (helfi92) [inline README.md]: Fix indentation.

File: README.md [hunk 25]
```diff
@@ -57,6 +57,52 @@ Install npm dependencies and start it up:
...
+
+  <Suggestions for how, and when, to ask for help -- mailing lists, irc channels, bug or issue comments, etc.>
+
+
+products:
+- <Bugzilla Product>
+- <Bugzilla Product>: [<Bugzilla Component 1>, <Bugzilla Component 2>]
+repositories:
+- <Organization Name>/<Repository Name> : <Label to get the issue>
+e.g:
```
Comment (helfi92) [inline README.md]: Remove this line.

File: README.md [hunk 26]
```diff
@@ -57,6 +57,52 @@ Install npm dependencies and start it up:
...
+
+  ## How Do I Get Help?
+
+  <Suggestions for how, and when, to ask for help -- mailing lists, irc channels, bug or issue comments, etc.>
+
+
+products:
+- <Bugzilla Product>
+- <Bugzilla Product>: [<Bugzilla Component 1>, <Bugzilla Component 2>]
+repositories:
```
Comment (helfi92) [inline README.md]: Remove this line.

File: README.md [hunk 27]
```diff
@@ -57,6 +57,52 @@ Install npm dependencies and start it up:
...
+- <Organization Name>/<Repository Name> : <Label to get the issue>
```
Comment (helfi92) [inline README.md]: Replace this line with:

`- <Organization Name>/<Repository Name> : <GitHub Label 1>`
`- <Organization Name>/<Repository Name> : <GitHub Label 2>`

File: README.md [hunk 28]
```diff
@@ -57,6 +57,52 @@ Install npm dependencies and start it up:
...
+e.g:
+repositories:
+- <Organization Name>/<Repository Name>: good-first-bug
+- <Organization Name>/<Repository Name>: easy-fix
```
Comment (helfi92) [inline README.md]: You can remove L99-101.

File: README.md [hunk 29]
```diff
@@ -57,6 +57,49 @@ Install npm dependencies and start it up:
...
+
+  ### How Do I Write the Code?
+
+  <Summary of the development and patch-submission process -- pull requests? patches on bugzilla? tests?>
+
+  ## How Do I Get Help?
+
+  <Suggestions for how, and when, to ask for help -- mailing lists, irc channels, bug or issue comments, etc.>
+
+
```
Comment (helfi92) [inline README.md]: Remove one of the two empty lines.

File: README.md [hunk 30]
```diff
@@ -57,6 +57,49 @@ Install npm dependencies and start it up:
...
+- <Bugzilla Product>
+- <Bugzilla Product>: [<Bugzilla Component 1>, <Bugzilla Component 2>]
+repositories:
+- <Organization Name>/<Repository Name> : <Github Label 1>
+- <Organization Name>/<Repository Name> : <Github Label 2>
+
+```
+_Note: The `introduction` field is to be rendered as markdown, allowing for bullet points, links and other simple formatting._
+
 ### Contributing
```
Comment (helfi92) [inline README.md]: Let's make this an H2. `## Contribution`

File: README.md [hunk 31]
```diff
@@ -100,7 +99,7 @@ repositories:
 ```
 _Note: The `introduction` field is to be rendered as markdown, allowing for bullet points, links and other simple formatting._
 
-### Contributing
+## Contribution
```
Comment (helfi92) [inline README.md]: My bad, the text should stay Contributing.

**Pull request merged**